app/src/main/java/org/thoughtcrime/securesms/SessionDialogBuilder.kt

```diff
@@ -30,20 +43,42 @@ annotation class DialogDsl
 @DialogDsl
 class SessionDialogBuilder(val context: Context) {
 
-    private val dp20 = toPx(20, context.resources)
-    private val dp40 = toPx(40, context.resources)
-    private val dp60 = toPx(60, context.resources)
+    val dp20 = toPx(20, context.resources)
+    val dp40 = toPx(40, context.resources)
+    val dp60 = toPx(60, context.resources)
 
     private val dialogBuilder: AlertDialog.Builder = AlertDialog.Builder(context)
 
     private var dialog: AlertDialog? = null
-    private fun dismiss() = dialog?.dismiss()
+    fun dismiss() = dialog?.dismiss()
 
     private val topView = LinearLayout(context)
         .apply { setPadding(0, dp20, 0, 0) }
         .apply { orientation = VERTICAL }
         .also(dialogBuilder::setCustomTitle)
-    private val contentView = LinearLayout(context).apply { orientation = VERTICAL }
+
+    // Maximum height of the dialog. This comes into play in the Open URL dialog where we
+    // might have a a very, very long URL (which we provide a ScrollView for) - but without
+    // limiting the height of the dialog, the WRAP_CONTENT part can allow the dialog to be
+    // super tall!
+    // Note: I eye-balled this DP value for the max dialog height to make it match the
+    // height of regular not-scrollable dialogs.
+    val maxHeightInDp = 137
+    val scale = context.resources.displayMetrics.density
+    val maxHeightInPx = (maxHeightInDp * scale + 0.5f).toInt()
+
+    val contentView = LinearLayout(context).apply {
+        orientation = VERTICAL
+
+        // Limit the max height of the dialog to the above value
+        viewTreeObserver.addOnGlobalLayoutListener {
+            if (height > maxHeightInPx) {
+                layoutParams.height = maxHeightInPx
+                requestLayout()
+            }
+        }
+    }
```

Comment:
This feels a little dodgy.
This also won't be taken into account in our Compose dialogs.
So the best way to deal with this is by limiting the content itself. 
The dialog should be able to naturally stretch. This will come into play with accessibility where some users might have increased fonts and display and a hardcoded height is a bad idea.
Instead we need to limit the content passed to our dialog where it's required.